### PR TITLE
Change TypeProps for Ideogram and add new tests

### PR DESCRIPTION
**Pull request by Bachibouzouk**
## About 
Impose values for certain props like orientation which should only accept "vertical" or "horizontal".

## Before merging
- [ ] I have gone through the [code review checklist](https://github.com/plotly/dash-component-boilerplate/blob/master/%7B%7Bcookiecutter.project_shortname%7D%7D/review_checklist.md)
- [ ] I have completed all of the [steps to take before merging](https://github.com/plotly/dash-bio/blob/master/.github/before_merging.md)
- [ ] I know how to [deploy to DDS](https://github.com/plotly/dash-bio/blob/master/.github/deploying_to_dds.md)

### CODE REVIEW

File: dash_bio/Molecule3dViewer.py [hunk 1]
```diff
@@ -17,18 +17,21 @@ class Molecule3dViewer(Component):
 - modelData (optional): The data that will be used to display the molecule in 3D
 The data will be in JSON format
 and should have two main dictionaries - atoms, bonds. modelData has the following type: dict containing keys 'atoms', 'bonds'.
-Those keys have the following types:
+Those keys have the following types:
```
Comment (mkcor) [inline dash_bio/Molecule3dViewer.py]: I'm sure you'll fix it later, but I think it shouldn't have landed in a commit in the first place...

Comment (Bachibouzouk) [reply dash_bio/Molecule3dViewer.py]: @mkcor - This comes from running `npm run build:all`, I plan to do a rebase of this PR, and maybe even break this PR in smaller ones and delete this one.

Comment (mkcor) [reply dash_bio/Molecule3dViewer.py]: Ok. Looks like I'll have to study what's happening in this `build` at some point... #214

File: src/lib/components/Ideogram.react.js [hunk 2]
```diff
@@ -308,25 +309,19 @@ Ideogram.propTypes = {
...
-    ancestors: PropTypes.object,
-
     /**
      * Layout of ideogram annotations.
      * One of "tracks", "histogram", or "overlay".
      *
      * "tracks": display annotations in tracks beside each chromosome.
      *
      * "histogram": display annotations in a histogram. Clusters annotations
-     * by location. Each cluster/bin is shown as a bar, the height of which represents
```
Comment (shammamah) [inline src/lib/components/Ideogram.react.js]: Why did this change?

File: src/lib/components/Ideogram.react.js [hunk 3]
```diff
@@ -308,25 +309,19 @@ Ideogram.propTypes = {
...
+     * by location. Each cluster/bin is shown as a  bar which height represents
      * the number of annotations on genomic range.
      *
      * "overlay": display annotations directly over chromosomes.
      */
-    annotationsLayout: PropTypes.string,
+    annotationsLayout: PropTypes.oneOf(['tracks', 'histogram', 'overlay']),
```
Comment (shammamah) [inline src/lib/components/Ideogram.react.js]: Nice!

File: src/lib/components/Ideogram.react.js (hunk 2 above)
Comment (Bachibouzouk) [reply src/lib/components/Ideogram.react.js]: I changed it back

File: src/lib/components/Ideogram.react.js [hunk 4]
```diff
@@ -13,6 +13,9 @@ import {omit} from 'ramda';
  * by side tracks to show important genomic data.
  *
  * Go here to see it in action: https://eweitz.github.io/ideogram/
+ * Reference of the component's props :
+ * https://github.com/eweitz/ideogram/blob/
+ * 770ce1fe5ae4f27dd435f9f99948118af88f8349/api.md
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]: Oh, so it wasn't line-wrapping, the URL is over two lines in the source. Why don't we keep it over one line?

File: src/lib/components/Ideogram.react.js [hunk 5]
```diff
@@ -450,10 +445,10 @@ Ideogram.propTypes = {
     chrWidth: PropTypes.number,
 
     /**
-     * A list of the names of chromosomes to 
-     * display. Useful for depicting a subset of the chromosomes in the genome, 
+     * A list of the names of chromosomes to
+     * display. Useful for depicting a subset of the chromosomes in the genome,
      * e.g. a single chromosome.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * e.g., a single chromosome.
```
And it looks like the above three lines should fit over two lines.

File: src/lib/components/Ideogram.react.js [hunk 6]
```diff
@@ -477,15 +472,23 @@ Ideogram.propTypes = {
     dataDir: PropTypes.string,
 
     /**
-     * This is a work in progess and will hopefully be fixed in future releases.
+     * Organism(s) to show chromosomes for. Supply organism's name as a string (e.g. "human") or
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * Organism(s) to show chromosomes for. Supply organism's name as a string (e.g., "human") or
```
Just because I'm going by the American style.

File: src/lib/components/Ideogram.react.js [hunk 7]
```diff
@@ -477,15 +472,23 @@ Ideogram.propTypes = {
...
+     * organism's NCBI Taxonomy ID (taxid, e.g. 9606) to display chromosomes from a single organism,
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * organism's NCBI Taxonomy ID (taxID, e.g., 9606) to display chromosomes from a single organism,
```
Just because I'm seeing "taxID" later on.

File: src/lib/components/Ideogram.react.js [hunk 8]
```diff
@@ -477,15 +472,23 @@ Ideogram.propTypes = {
...
+     * or an array of organisms' names or taxids to display chromosomes from multiple species.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * or an array of organisms' names or taxIDs to display chromosomes for all these organisms.
```
It might be obvious that "species" and "organisms" mean the same thing in this context, but let's not introduce confusion. Actually, it seems to me that "species" would be more accurate all the way... I see an organism as an instance of a given species.

File: src/lib/components/Ideogram.react.js [hunk 9]
```diff
@@ -477,15 +472,23 @@ Ideogram.propTypes = {
...
+    localOrganism: PropTypes.object,
 
     /**
      * Used to compare two chromosomes with each other.
-     * The keys "chrOne" and "chrTwo" represent one chromosome each. Organism is the 
-     * taxID or name. Start is an array, containing start one and 
-     * start two in this order. Stop is array, containing stop one, and stop two, 
+     * The keys "chrOne" and "chrTwo" represent one chromosome each. Organism is the
+     * taxID or name. Start is an array, containing start one and
+     * start two in this order. Stop is array, containing stop one, and stop two,
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * start two in this order. Stop is array, containing stop one and stop two,
```
It seems that all these words would want quotes or back ticks, since they refer to keys or variable names, just like "chrOne" and "chrTwo"...

File: src/lib/components/Ideogram.react.js [hunk 10]
```diff
@@ -526,47 +529,21 @@ Ideogram.propTypes = {
...
     /**
-     * Whether annotations should be filterable.
-     */
-    filterable: PropTypes.number,
-
-    /**
-     * Provide local JSON organism into this prop from a local user JSON file.
-     * DataDir must not be initialized.
+     * The resolution of cytogenetic bands to show for each chromosome.
+     * The quantity refers to approximate value in bands per haploid set (bphs).
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * The quantity refers to an approximate value in bands per haploid set (bphs).
```

File: src/lib/components/Ideogram.react.js [hunk 11]
```diff
@@ -526,47 +529,21 @@ Ideogram.propTypes = {
...
+     * One of 450, 550, or 850.
      */
-    localOrganism: PropTypes.object,
+    resolution: PropTypes.number,
 
     /**
-     * Dash event callback for mousing over data.
+     * Whether annotations should be filterable.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * Whether annotations should be filterable or not.
```

File: src/lib/components/Ideogram.react.js [hunk 12]
```diff
@@ -580,6 +557,12 @@ Ideogram.propTypes = {
      */
     ploidyDesc: PropTypes.arrayOf(PropTypes.object),
 
+    /**
+     *  A map associating ancestor labels to colors. Used to color
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * A map associating ancestor labels to colors. Used to color
```

File: src/lib/components/Ideogram.react.js [hunk 13]
```diff
@@ -589,34 +572,19 @@ Ideogram.propTypes = {
     /**
      * Whether chromosomes are rotatable on click.
      */
-
     rotatable: PropTypes.bool,
 
     /**
      * Dash callback that returns True if rotated, and false if not.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * Dash callback that returns true if rotated, and false if not.
```
Python vs. JS...

File: src/lib/components/Ideogram.react.js [hunk 14]
```diff
@@ -641,4 +609,43 @@ Ideogram.propTypes = {
...
+     */
+
+    /**
+     * Callback function to invoke when brush moves.
+     * onBrushMove: PropTypes.func,
+     */
+
+    /**
+     * Callback function to invoke when annotations are drawn. (React)
+     *  onDrawAnnots: PropTypes.func,
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * onDrawAnnots: PropTypes.func,
```

File: src/lib/components/Ideogram.react.js [hunk 15]
```diff
@@ -641,4 +609,43 @@ Ideogram.propTypes = {
...
+     */
+
+    /**
+     * Callback function to invoke when chromosomes are loaded,
+     * i.e. rendered on the page. (React)
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * i.e., rendered on the page. (React)
```

File: src/lib/components/Ideogram.react.js [hunk 16]
```diff
@@ -641,4 +609,43 @@ Ideogram.propTypes = {
...
+     * onLoad: PropTypes.func,
+     */
+
+    /**
+     * Useful for putting ideogram into a small container,
+     * or when dealing with genomes that have many chromosomes.
+     * Note: Not fully working needs to be fixed by developer.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * Note: Not fully working, needs to be fixed by developer.
```

File: dash_bio/metadata.json [hunk 17]
```diff
@@ -1684,30 +1700,22 @@
...
+            },
+            {
+              "value": "'female'",
+              "computed": false
+            }
+          ]
         },
         "required": false,
-        "description": "Useful for omitting chromosome Y in female mammals.\nCurrently only supported for organisms that use XY sex-determination.\nAll string values will show chromosome Y, only the value 'female' will hide it"
+        "description": "Useful for omitting chromosome Y in female mammals.\nCurrently only supported for organisms that use XY sex-determination."
```
Comment (mkcor) [inline dash_bio/metadata.json]: For example, here, I think "species" would be more fitting than "organisms." What do you think @NadiaTahiri ?

File: dash_bio/metadata.json [hunk 18]
```diff
@@ -1183,7 +1183,7 @@
     }
   },
   "src/lib/components/Ideogram.react.js": {
-    "description": "The Dash Ideogram is used to draw and animate genome-wide\ndatasets for organisms such as an human, mouse, and any\nother eukaryotes. The Ideogram can be used to compare\nhomologous features between chromosomes, and depict,\nhaploid, diploid, aneuploidy genomes. It can also display\nannotations using histograms, overlays, and simple side\nby side tracks to show important genomic data.\n\nGo here to see it in action: https://eweitz.github.io/ideogram/",
+    "description": "The Dash Ideogram is used to draw and animate genome-wide\ndatasets for organisms such as an human, mouse, and any\nother eukaryotes. The Ideogram can be used to compare\nhomologous features between chromosomes, and depict,\nhaploid, diploid, aneuploidy genomes. It can also display\nannotations using histograms, overlays, and simple side\nby side tracks to show important genomic data.\n\nGo here to see it in action: https://eweitz.github.io/ideogram/\nReference of the component's props :\nhttps://github.com/eweitz/ideogram/blob/\n770ce1fe5ae4f27dd435f9f99948118af88f8349/api.md",
```
Comment (mkcor) [inline dash_bio/metadata.json]:
```suggestion
    "description": "The Dash Ideogram is used to draw and animate genome-wide\ndatasets for species such as human, mouse, and any\nother eukaryotes. The Ideogram can be used to compare\nhomologous features between chromosomes, and depict,\nhaploid, diploid, aneuploidy genomes. It can also display\nannotations using histograms, overlays, and simple side\nby side tracks to show important genomic data.\n\nTo see it in action, visit: https://eweitz.github.io/ideogram/\nReference of the component's props :\nhttps://github.com/eweitz/ideogram/blob/\n770ce1fe5ae4f27dd435f9f99948118af88f8349/api.md",
```

File: src/lib/components/Ideogram.react.js (hunk 4 above)
Comment (Bachibouzouk) [reply src/lib/components/Ideogram.react.js]: @mkcor Because it is more than 100 characters then

File: dash_bio/metadata.json (hunk 18 above)
Comment (Bachibouzouk) [reply dash_bio/metadata.json]: @mkcor this file is autogenerated, I will change the source in the .react.js :)

File: src/lib/components/Ideogram.react.js (hunk 4 above)
Comment (mkcor) [reply src/lib/components/Ideogram.react.js]: Isn't it 91 though?

File: dash_bio/metadata.json (hunk 18 above)
Comment (mkcor) [reply dash_bio/metadata.json]: Oh! I Ctrl-F'd "an human" and didn't find it. Ok.

File: src/lib/components/Ideogram.react.js (hunk 4 above)
Comment (Bachibouzouk) [reply src/lib/components/Ideogram.react.js]: Yes it is, it was because my PyCharm switched to 80 line wrap :)

File: src/lib/components/Ideogram.react.js (hunk 6 above)
Comment (Bachibouzouk) [reply src/lib/components/Ideogram.react.js]: I replaced it everywhere ;)

**Comment by Bachibouzouk** [comment]
@mkcor : I implemented your comments (except replacing organisms by species, I'll wait for @NadiaTahiri 's answer for that :)

**Comment by mkcor** [comment]
> @mkcor : I implemented your comments (except replacing organisms by species, I'll wait for @NadiaTahiri 's answer for that :)

Wonderful! I'm re-running the CI workflow to get the build to pass... But I'm noticing this fail ("assert doubling of the 22 chromosomes + X and Y chromosomes" in Ideogram) over and over again, so we might have a robustness issue there.

File: src/lib/components/Ideogram.react.js [hunk 19]
```diff
@@ -288,8 +288,8 @@ Ideogram.defaultProps = {
 
 Ideogram.propTypes = {
     /**
-     * The ID used to identify this component in Dash callbacks
-     * and used to identify Ideogram instances.
+     * The ID used to identify this component in Dash callbacks and used to identify Ideogram
+     *instances.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * instances.
```

File: src/lib/components/Ideogram.react.js [hunk 20]
```diff
@@ -450,55 +441,60 @@ Ideogram.propTypes = {
...
         PropTypes.object,
     ]),
 
     /**
-     * Absolute or relative URL of the directory
-     * containing data needed to draw banded chromosomes.
-     * You will need to set up you're own database to grab data from
-     * for custom data.
+     * Absolute or relative URL of the directory containing data needed to draw banded chromosomes.
+     * You will need to set up you're own database to grab data from for custom data.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * You will need to set up your own database to grab data from a custom database.
```

File: src/lib/components/Ideogram.react.js [hunk 21]
```diff
@@ -517,128 +513,127 @@ Ideogram.propTypes = {
...
+     * Dash callback that returns True if rotated, and false if not.
+     *
      */
-    ploidy: PropTypes.number,
+    rotated: PropTypes.bool,
 
     /**
-     * Description of ploidy in each chromosome set in terms of
-     * ancestry composition.
+     * Useful for omitting chromosome Y in female mammals.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]: Apparently, the XY sex-determination system exists also in species which are not mammals! /cc @NadiaTahiri

File: src/lib/components/Ideogram.react.js [hunk 22]
```diff
@@ -517,128 +513,127 @@ Ideogram.propTypes = {
...
-     */
+     * Whether to show cytogenetic band labels, e.g., 1q21
+     **/
+    showBandLabels: PropTypes.bool,
 
-    rotatable: PropTypes.bool,
+    /**
+     * Whether to show fully banded chromosomes for genomes that have sufficient data. Useful for
+     * showing simpler chromosomes of cytogenetically well-characterized organisms, e.g., human,
+     * beside chromosomes of ess studied organisms, e.g., chimpanzee.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * beside chromosomes of less studied organisms, e.g., chimpanzee.
```

File: src/lib/components/Ideogram.react.js [hunk 23]
```diff
@@ -517,128 +513,127 @@ Ideogram.propTypes = {
...
+     * Callback function to invoke after chromosome has rotated. (React)
+     * onDidRotate: PropTypes.func,
      */
-    resolution: PropTypes.number,
 
     /**
-     * Useful for putting ideogram into a small container,
-     * or when dealing with genomes that have many chromosomes.
-     * Note: Not fully working needs to be fixed by developer.
+     * Dash event callback for mousing over data.
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
     * Dash event callback for hovering over data.
```

File: src/lib/components/Ideogram.react.js (hunk 6 above)
Comment (mkcor) [reply src/lib/components/Ideogram.react.js]: It holds for *i.e.* as well, see https://github.com/plotly/dash-bio/pull/201#discussion_r263813029 ;)

File: src/lib/components/Ideogram.react.js [hunk 24]
```diff
@@ -13,6 +13,8 @@ import {omit} from 'ramda';
  * by side tracks to show important genomic data.
  *
  * Go here to see it in action: https://eweitz.github.io/ideogram/
+ * Reference of the component's props :
```
Comment (mkcor) [inline src/lib/components/Ideogram.react.js]:
```suggestion
 * Reference for the component's props:
```

File: tests/test_common_features.py [hunk 25]
```diff
@@ -116,6 +117,12 @@ def create_test_layout(app_name, component_base, **kwargs):
     )
 
 
+def clear_field(web_elem):
+    """Analog to web_elem.clear() which does not work with Chrome currently."""
```
Comment (mkcor) [inline tests/test_common_features.py]:
```suggestion
    """Do the same as web_elem.clear() but work with Chrome."""
```
(Would be "analog of" anyway.)

File: tests/test_common_features.py [hunk 26]
```diff
@@ -116,6 +117,12 @@ def create_test_layout(app_name, component_base, **kwargs):
...
+    web_elem.send_keys(Keys.CONTROL + "a")
```
Comment (mkcor) [inline tests/test_common_features.py]:
```suggestion
    web_elem.send_keys(Keys.CONTROL + 'a')
```

File: dash_bio/metadata.json (hunk 18 above)
Comment (mkcor) [reply dash_bio/metadata.json]: @Bachibouzouk [source](https://github.com/plotly/dash-bio/blob/0389fb0e512cd323d61e8eca00a7b81eaab306ec/src/lib/components/Ideogram.react.js#L8) still reads "an human," please fix!

File: src/lib/components/Ideogram.react.js (hunk 14 above)
Comment (Bachibouzouk) [reply src/lib/components/Ideogram.react.js]: I will skip this one as this prop will probably not be implemented as it is to feed a react function to the component, so it is likely to be erased in the future :)

File: CHANGELOG.md [hunk 27]
```diff
@@ -4,9 +4,13 @@
...
 ### Removed
 * Removed mentions of Dash events in OncoPrint component.
+* Removed properties which weren't used in Ideogram component.
 
 ### Changed
-* Changed property `fullChromosomeLabels` so that it can be updated using dash callbacks
+* Changed property `fullChromosomeLabels` in Ideogram component so that it can be updated using 
+dash callbacks.
+* Restricted possible values for certain properties in Ideogram 
+component to a finite set of values.
```
Comment (mkcor) [inline CHANGELOG.md]: It's not clear to me why lines wrap at such different lengths...? Line 10 vs line 12.

File: dash_bio/Ideogram.py [hunk 28]
```diff
@@ -14,15 +14,16 @@ class Ideogram(Component):
 by side tracks to show important genomic data.
 
 Go here to see it in action: https://eweitz.github.io/ideogram/
+Reference of the component's props :
+https://github.com/eweitz/ideogram/blob/
+770ce1fe5ae4f27dd435f9f99948118af88f8349/api.md
```
Comment (mkcor) [inline dash_bio/Ideogram.py]: It doesn't look like you regenerated file `Ideogram.py` after commit 0a7c4dc87c1929c8f.

File: src/lib/components/Ideogram.react.js (hunk 21 above)
Comment (TahiriNadia) [reply src/lib/components/Ideogram.react.js]: The flower represents the reproductive organ in flowering plants (_i.e._ _angiosperms_). 
The stamens produce the pollen grains that contain the **male** reproductive cells, 
while the pistil is where the ovules are produced, so **female** reproductive
_e.g._ Fertilization is usually cross-fertilized, that is, in most cases, a plant can not self-pollinate.
Source: [wikipedia](https://en.wikipedia.org/wiki/Plant_reproductive_morphology)

In conclusion, we can't talk X/Y sex determination system like animal side because is correlated to the organ 
@TahiriNadia here in my GitHub account which I use the other one is my old version ;-)

Comment (TahiriNadia) [reply src/lib/components/Ideogram.react.js]: And **Bacteria** is most complex because they may exchange genetic information

Comment (mkcor) [reply src/lib/components/Ideogram.react.js]: Thanks, @TahiriNadia! I wasn't suggesting we should include plants; there are *animals* which are not *mammals*. Would it make more sense to write "female animals" rather than "female mammals"?

Comment (TahiriNadia) [reply src/lib/components/Ideogram.react.js]: Yes, you definitively can use `female animals` rather than `female mammals` is more strict on this second case.

Comment (TahiriNadia) [reply src/lib/components/Ideogram.react.js]: And the formal notation is `specie` instead of `organism`

Comment (mkcor) [reply src/lib/components/Ideogram.react.js]: Thanks for sharing your expertise, @TahiriNadia! We'll include these improvements in a subsequent PR.

Comment (TahiriNadia) [reply src/lib/components/Ideogram.react.js]: You're welcome @mkcor
